packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -725,14 +725,24 @@ class RenderParagraph extends RenderBox with ContainerRenderObjectMixin<RenderBo
   @override
   bool hitTestChildren(BoxHitTestResult result, { required Offset position }) {
     final TextPosition textPosition = _textPainter.getPositionForOffset(position);
-    final Object? span = _textPainter.text!.getSpanForPosition(textPosition);
-    if (span is HitTestTarget) {
-      result.add(HitTestEntry(span));
-      return true;
+    if (_positionIsWithinText(position: position, textPosition: textPosition)) {
+      final Object? span = _textPainter.text!.getSpanForPosition(textPosition);
```

Comment:
weird it is declared as `Object?` it is probably a mistake I think

Reply:
done